src/kotlin/kage/Age.kt

```diff
@@ -105,22 +106,24 @@ public object Age {
       throw NoRecipientsException("No recipients specified")
     }
 
-    // From the age docs:
-    // As a best effort, prevent an API user from generating a file that the
-    // ScryptIdentity will refuse to decrypt. This check can't unfortunately be
-    // implemented as part of the Recipient interface, so it lives as a special
-    // case in Encrypt.
-    // https://github.com/FiloSottile/age/blob/ab3707c085f2c1fdfd767a2ed718423e3925f4c4/age.go#L114-L122
-    recipients.forEach { recipient ->
-      if (recipient is ScryptRecipient && recipients.size != 1) {
-        throw InvalidScryptRecipientException("Only one scrypt recipient is supported")
+    val fileKey = generateFileKey()
+    var stanzas = mutableListOf<AgeStanza>()
+    var labels = emptyList<String>()
+
+    for (idx in 0 until recipients.size) {
+      val recipient = requireNotNull(recipients[idx])
+      val (s, l) = wrapWithLabels(recipient, fileKey)
+      val sorted = l.sorted()
+      if (idx == 0) {
+        labels = sorted
+      } else {
+        if (labels != sorted) {
+          throw InvalidScryptRecipientException("incompatible scrypt recipients")
+        }
       }
```

Comment:
nit: we should probably add a comment to explain what's happening here. 

Also, I would prefer doing this instead. It makes the intent a little bit more clearer imo

```kotlin
if (idx == 0) {
  labels = sorted
  continue
}

if (labels != sorted) {
  throw InvalidScryptRecipientException("incompatible scrypt recipients")
}
```